mcux/boards/CMakeLists.txt

```diff
@@ -23,11 +23,12 @@ endif()
 zephyr_compile_definitions_ifdef(CONFIG_NXP_IMX_RT_BOOT_HEADER XIP_BOOT_HEADER_ENABLE=1)
 zephyr_compile_definitions_ifdef(CONFIG_NXP_IMX_RT6XX_BOOT_HEADER BOOT_HEADER_ENABLE=1)
 zephyr_compile_definitions_ifdef(CONFIG_DEVICE_CONFIGURATION_DATA XIP_BOOT_HEADER_DCD_ENABLE=1)
-zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE)
 
 if (${MCUX_BOARD} MATCHES "evk[bm]imxrt1[0-9][0-9][0-9]")
 zephyr_library_sources_ifdef(CONFIG_BOOT_FLEXSPI_NOR ${MCUX_BOARD}/${MCUX_BOARD}_flexspi_nor_config.c)
 zephyr_library_sources_ifdef(CONFIG_DEVICE_CONFIGURATION_DATA ${MCUX_BOARD}/${MCUX_BOARD}_sdram_ini_dcd.c)
+zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE)
 elseif (${MCUX_BOARD} MATCHES "evkmimxrt6[0-9][0-9]")
 zephyr_library_sources_ifdef(CONFIG_NXP_IMX_RT6XX_BOOT_HEADER ${MCUX_BOARD}/flash_config.c)
+zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024)
```

Comment:
Does the sdk really interpret the same symbol differently for rt600 vs. rt10xx?

Reply:
Yes, for MXRT685 it is used here
https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/boards/evkmimxrt685/flash_config.c#L61

This values needs to be in bytes.